### DISCUSSION TITLE
Close `ModalScreen` on click outside modal content

### DISCUSSION THREAD

**Discussion by TomJGooding** [Textualize/textual#4483] [General]
You might sometimes want to close a `ModalScreen` by simply clicking outside the modal content (i.e. the darkened background).

I couldn't find anything in the docs or other discussions, so just sharing this in case it helps someone.

Please let me know if there is a better way!

```python
    def on_click(self, event: events.Click) -> None:
        clicked, _ = self.get_widget_at(event.screen_x, event.screen_y)
        # Close the screen if the user clicks outside the modal content
        # (i.e. the darkened background)
        if clicked is self:
            self.app.pop_screen()
```

**Comment by davep** [comment]
> Please let me know if there is a better way!

[Me too](https://github.com/Textualize/textual/blob/17e975db679bbe15877a57ac7a9390260a6ee386/src/textual/command.py#L615-L626).

**Reply by TomJGooding** [reply]
Thanks Dave, I thought I'd done a pretty thorough search for other examples but somehow missed that one!

**Reply by davep** [reply]
Easily missed though, hardly a standout example. I did write another example some time ago but I suspect that was as an illustration for someone on the Textualize Discord server: https://github.com/davep/textual-sandbox/blob/main/attic/click_outside_modal.py

**Comment by willmcgugan** [comment]
Looks like a solid solution to me!
